echo_handler.cc

```diff
@@ -1,10 +1,18 @@
 #include "echo_handler.h"
 
-void EchoHandler::handle_request(Request* req, Response* rep) {
-  rep->http_version = req->GetVersion();
-  rep->status = Response::ok;
-  rep->headers["Content-Type"] = "text/plain";
-  std::size_t body_length = req->GetOriginalString().find("\r\n\r\n") + size_t(4);
-  rep->headers["Content-Length"] = std::to_string((int) body_length);
-  rep->body = req->GetOriginalString();
+RequestHandler::Status EchoHandler::Init(const std::string& uri_prefix,
+                                         const NginxConfig& config)
+{
+  return RequestHandler::Status::OK;
+}
+
+RequestHandler::Status EchoHandler::HandleRequest(const Request& request,
+                                                  Response* response)
+{
+  response->SetStatus(Response::ok);
+  response->AddHeader("Content-Type", "text/plain");
+  std::size_t body_length = request.raw_request().find("\r\n\r\n") + size_t(4);
```

Comment:
Not totally sure why the find thing is necessary, might be an issue for requests with bodies (if that's something we're gonna need to do)